yakut/param/formatter.py

```diff
@@ -94,12 +175,65 @@ def _unittest_formatter() -> None:
+    tsvh_formatter = _FORMATTERS["TSVH"]()
+    # first time should include a header
+    assert tsvh_formatter(obj) == "2345.abc.def.[0]\t2345.abc.def.[1]\t2345.ghi\n123\t456\t789"
+    # subsequent calls shouldn't include a header
+    assert tsvh_formatter(obj) == "123\t456\t789"
+    from decimal import Decimal
+    from math import nan
+
+    obj = {
+        142: {
+            "_metadata_": {
+                "timestamp": {"system": Decimal("1640610921.414715"), "monotonic": Decimal("4522.612870")},
+                "priority": "nominal",
+                "transfer_id": 17,
+                "source_node_id": 21,
+            },
+            "timestamp": {"microsecond": 66711825},
+            "value": {
+                "kinematics": {
+                    "angular_position": {"radian": nan},
+                    "angular_velocity": {"radian_per_second": 375860.15625},
+                    "angular_acceleration": {"radian_per_second_per_second": 0.0},
+                },
+                "torque": {"newton_meter": nan},
+            },
+        }
+    }
+    assert _FORMATTERS["TSV"]()(obj) == "1640610921.414715	4522.612870	nominal	17	21	66711825	nan	375860.15625	0.0	nan"
+    obj = {
+        142: {
+            "_metadata_": {
+                "timestamp": {"system": Decimal("1640611164.396007"), "monotonic": Decimal("4765.594161")},
+                "priority": "nominal",
+                "transfer_id": 28,
+                "source_node_id": 21,
+            },
+            "timestamp": {"microsecond": 309697890},
+            "value": {
+                "kinematics": {
+                    "angular_position": {"radian": nan},
+                    "angular_velocity": {"radian_per_second": 0.0},
+                    "angular_acceleration": {"radian_per_second_per_second": 0.0},
+                },
+                "torque": {"newton_meter": nan},
+            },
+        }
+    }
+    tsvfc_formatter = _FORMATTERS["TSVFC"]()
+    assert (
+            tsvfc_formatter(obj)
+            == "142{	142._metadata_{	142._metadata_.timestamp{	142._metadata_.timestamp.system	142._metadata_.timestamp.monotonic	142._metadata_.timestamp}	142._metadata_.priority	142._metadata_.transfer_id	142._metadata_.source_node_id	142._metadata_}	142.timestamp{	142.timestamp.microsecond	142.timestamp}	142.value{	142.value.kinematics{	142.value.kinematics.angular_position{	142.value.kinematics.angular_position.radian	142.value.kinematics.angular_position}	142.value.kinematics.angular_velocity{	142.value.kinematics.angular_velocity.radian_per_second	142.value.kinematics.angular_velocity}	142.value.kinematics.angular_acceleration{	142.value.kinematics.angular_acceleration.radian_per_second_per_second	142.value.kinematics.angular_acceleration}	142.value.kinematics}	142.value.torque{	142.value.torque.newton_meter	142.value.torque}	142.value}	142}\n{	{	{	1640611164.396007	4765.594161	}	nominal	28	21	}	{	309697890	}	{	{	{	nan	}	{	0.0	}	{	0.0	}	}	{	nan	}	}	}"
```

Comment:
wowza that's a long line. Can you please split it like `"\t".join(["142{", "142._metadata_{", ...])`

Reply:
why?

Reply:
also no

Reply:
Because it's long and does not conform to the [conventions](https://kb.zubax.com/display/MAINKB/Zubax+Python+coding+conventions).

Reply:
done